ecs-agent/wsclient/client_test.go

```diff
@@ -0,0 +1,399 @@
+	err := cs.Connect()
+
+	assert.Error(t, err, "Expected error for incorrect URL scheme")
+}
+
+// TestWebsocketScheme checks that websocketScheme handles valid and invalid mappings
+// correctly
+func TestWebsocketScheme(t *testing.T) {
+	// test valid schemes
+	validMappings := map[string]string{
+		"http":  "ws",
+		"https": "wss",
+	}
+
+	for input, expectedOutput := range validMappings {
+		actualOutput, err := websocketScheme(input)
+
+		assert.NoError(t, err, "Unexpected error for valid http scheme")
+		assert.Equal(t, actualOutput, expectedOutput, "Valid http schemes should map to a websocket scheme")
+	}
+
+	// test an invalid mapping
+	_, err := websocketScheme("highly-likely-to-be-junk")
+	assert.Error(t, err, "Expected error for invalid http scheme")
+}
+
+func TestSetReadDeadlineClosedConnection(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	conn := mock_wsconn.NewMockWebsocketConn(ctrl)
+	cs := &ClientServerImpl{conn: conn}
+
+	ctx := context.Background()
+	opErr := &net.OpError{Err: errors.New(errClosed)}
+	conn.EXPECT().SetReadDeadline(gomock.Any()).Return(opErr)
+	assert.EqualError(t, cs.ConsumeMessages(ctx), opErr.Error())
+}
+
+func TestSetReadDeadlineError(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	conn := mock_wsconn.NewMockWebsocketConn(ctrl)
+	cs := &ClientServerImpl{conn: conn}
+	ctx := context.Background()
+	gomock.InOrder(
+		conn.EXPECT().SetReadDeadline(gomock.Any()).Return(errors.New("error")),
+		conn.EXPECT().SetWriteDeadline(gomock.Any()).Return(nil),
+		conn.EXPECT().Close().Return(nil),
+	)
+	assert.Error(t, cs.ConsumeMessages(ctx))
+}
+
+func TestAddRequestPayloadHandler(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	conn := mock_wsconn.NewMockWebsocketConn(ctrl)
+	conn.EXPECT().SetReadDeadline(gomock.Any()).Return(nil).MinTimes(1)
+	conn.EXPECT().ReadMessage().Return(websocket.TextMessage,
+		[]byte(`{"type":"PayloadMessage","message":{"tasks":[{"arn":"arn"}]}}`),
+		nil).MinTimes(1)
+	conn.EXPECT().SetWriteDeadline(gomock.Any()).Return(nil)
+	conn.EXPECT().Close()
+
+	closeWS := make(chan []byte)
+	defer close(closeWS)
+
+	ctx := context.Background()
+	mockServer, _, _, _, _ := utils.GetMockServer(closeWS)
+	mockServer.StartTLS()
+
+	types := []interface{}{ecsacs.PayloadMessage{}}
+	messageError := make(chan error)
+	cs := getClientServer(mockServer.URL, types)
+	cs.conn = conn
+
+	defer cs.Close()
+
+	messageChannel := make(chan *ecsacs.PayloadMessage)
+	reqHandler := func(payload *ecsacs.PayloadMessage) {
+		messageChannel <- payload
+	}
+	cs.AddRequestHandler(reqHandler)
+
+	go func() {
+		messageError <- cs.ConsumeMessages(ctx)
+		cs.Close()
+	}()
+
+	expectedMessage := &ecsacs.PayloadMessage{
+		Tasks: []*ecsacs.Task{{
+			Arn: aws.String("arn"),
+		}},
+	}
+
+	assert.Equal(t, expectedMessage, <-messageChannel)
+
+}
+
+//test for DecodeConnectionError
+//use WriteCloseMessage
+
+//send httpResponse in Connect()
```

Comment:
Are there missing tests here on L349-L352, or what are these comments?

Reply:
Missed removing the comments. I was using it while writing new tests to ensure the coverage requirements are met. Will remove them in latest commit.